_test/test_unique_objects_container/test_unique_references.m

```diff
@@ -28,7 +28,147 @@ function delete(~)
+            unique_references_container.global_container( ...
+                'CLEAR','multifit_issue');
+
+
+            urr = serializable.from_struct(savestr);
+            glr = unique_references_container.global_container( ...
+                'value','multifit_issue');
+            % forgets after reloading new state
+            assertTrue(isa(glr,'unique_objects_container'))
+            assertEqual(glr.n_objects,1)
+            assertEqual(glr.n_unique,1)
+            assertEqual(glr.unique_objects,{'bbb'})
+
+            savestr_cp = urr.to_struct();
+            % need to compare structures as initial urc is broken after
+            % clear
+            assertEqual(savestr,savestr_cp);
+            clear clSt;
+        end
+
+        function test_save_restore_global_state_two_urcs(~)
+            clWa = set_temporary_warning('off','HERBERT:unique_references_container:debug_only_argument');
+            function urc_clearer()
+                unique_references_container.global_container('CLEAR','multifit_issue')
+            end
+            clSt  = onCleanup(@urc_clearer);
+
+            urc1 = unique_references_container('multifit_issue','char');
+            urc1(1) = 'aaa';
+            urc1(2) = 'aaa';
+            urc1(3) = 'aaa';
+            urc1(4) = 'bbb';
+            urc2 = unique_references_container('multifit_issue','char');
+            urc2(1) = 'ccc';
+            urc2(2) = 'ccc';
+            urc2(3) = 'ccc';
+            urc2(4) = 'bbb';
+
+
+            glc = unique_references_container.global_container( ...
+                'value','multifit_issue');
+            assertTrue(isa(glc,'unique_objects_container'))
+            assertEqual(glc.n_objects,3)
+            assertEqual(glc.n_unique,3)
+            assertEqual(glc.unique_objects,{'aaa','bbb','ccc'})
+
+
+            savestr1_or = urc1.to_struct();
+            savestr2_or = urc2.to_struct();
+            unique_references_container.global_container( ...
+                'CLEAR','multifit_issue');
+
 
+            urr1 = serializable.from_struct(savestr1_or);
+            urr2 = serializable.from_struct(savestr2_or);
+
+            glr = unique_references_container.global_container( ...
+                'value','multifit_issue');
+            assertTrue(isa(glr,'unique_objects_container'))
+            assertEqual(glr.n_objects,3)
+            assertEqual(glr.n_unique,3)
+            assertEqual(glr.unique_objects,{'aaa','bbb','ccc'})
+
+
+            savestr_copy1 = urr1.to_struct();
+            savestr_copy2 = urr2.to_struct();
+
+            % need to compare structures as initial urc is broken after
+            % clear
+            assertEqual(savestr1_or,savestr_copy1);
+            assertEqual(savestr2_or,savestr_copy2);
+            clear clSt;
+        end
+
+        function test_save_restore_global_state(~)
+            clWa = set_temporary_warning('off','HERBERT:unique_references_container:debug_only_argument');
```

Comment:
Still not sure what this is testing separately to the 1st one.

Reply:
operations after replacemement may go different way (codewise). Not in Chris case, but in my alternative solution it happens for efficiency. 

This is the difference, so this test checks alternative internal code path.